### PR TITLE
Fix confidence ensembles RNNT logprobs selection logic for exclude_blank scenario

### PR DESCRIPTION
**Pull request by KunalDhawan**
# What does this PR do ?

- Fixes the algorithm for filtering blank frames while building confidence ensembles
- Adds checks for confidence value calculation to handle the edge case of all blank frames

**Collection**: [Note which collection this PR will affect]
ASR 

# Changelog 
- changes to get_filtered_logprobs() and compute_confidence() functions in nemo/collections/asr/models/confidence_ensembles.py

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

### CODE REVIEW

File: nemo/collections/asr/models/confidence_ensemble.py [hunk 1]
```diff
@@ -135,7 +136,10 @@ def compute_confidence(hypothesis: Hypothesis, confidence_cfg: ConfidenceConfig)
         alpha = confidence_cfg.method_cfg.temperature
     conf_func = get_confidence_measure_bank()[conf_type]
 
-    conf_value = aggr_func(conf_func(filtered_logprobs, v=vocab_size, t=alpha)).cpu().item()
+    if filtered_logprobs.shape[0] == 0:  # for the edge-case of all blanks, no frames
```
Comment (Kipok) [inline nemo/collections/asr/models/confidence_ensemble.py:139]: I'm not sure that returning 0 is the best choice here. The model might be very confident in predicting only blanks when there is a silence and we probably shouldn't ignore that. So maybe safer to use the same approach as for rnnt and return a confidence of a single blank token. Something like this in the `get_filtered_logprobs` function
```
            if filtered_logprobs.shape[0] == 0:  # for the edge-case of all blanks
                filtered_logprobs = logprobs[:1]
```

**Review by Kipok — CHANGES_REQUESTED** [review]
Just a small issue to fix for consistency with rnnt logic.

By the way, were you able to figure out why rnnt probabilities are not correctly normalized?

File: nemo/collections/asr/models/confidence_ensemble.py (hunk 1 above)
Comment (KunalDhawan) [reply nemo/collections/asr/models/confidence_ensemble.py]: Good point! Fixed this issue in the latest commit by following a similar approach for CTC as done for RNNT in get_filtered_logprobs(). 
Still investigating the rnnt normalization issue of why log_normalize isn't being propagated as true

**Review by Kipok — APPROVED** [review]
Thanks!

**Pull request merged**